src/GreenPlayer.js

```diff
@@ -198,6 +205,13 @@ class SoundPlayer extends EventEmitter {
 
         this.isPlaying = true;
 
+        const isStarting = this.isStarting = Promise.resolve()
```

Comment:
hrm... part of me thinks it might be better to just compare the lastStartTime to audioContext.currentTime -- this kind of depends on the microtasks running in the right order and might still emit a bunch of start events at the same audio context time?

Reply:
This needs to be aware of the javascript context's state, has the rest of the main thread performed work. This is to keep users from playing a possibly infinite number of copies of a sound.

This could test both things but I don't think it can work only by testing time. The time it cares about is at the millisecond scale anyways.

(We may want to make a separate microsecond check to determine if we should take and stop or take and stopImmediately.)